tests/test_common.py

```diff
@@ -265,34 +271,151 @@ def get(self, key: str, range_end: str = "", rev: int = 0):
+        req.value = value.encode("utf-8")
+        j = MessageToDict(req)
+        res = self.client.post("/v3/kv/put", json=j)
+        check_response(res)
+        if wait_for_commit:
+            rev, term = extract_rev_term(res)
+            self.wait_for_commit(term, rev)
+        if check_receipt:
+            res_pb = ParseDict(res.json(), etcd_pb2.PutResponse())
+            self.check_receipt("put", req, res_pb)
+        return res
+
+    def delete(
+        self,
+        key: str,
+        range_end: str = "",
+        wait_for_commit: bool = True,
+        check_receipt=True,
+    ):
         """
         Perform a delete operation on lskv.
         """
         logger.info("Delete: {} {}", key, range_end)
-        j = {"key": b64encode(key)}
+        req = etcd_pb2.DeleteRangeRequest()
+        req.key = key.encode("utf-8")
         if range_end:
-            j["range_end"] = b64encode(range_end)
-        return self.client.post("/v3/kv/delete_range", json=j)
+            req.range_end = range_end.encode("utf-8")
+        j = MessageToDict(req)
+        res = self.client.post("/v3/kv/delete_range", json=j)
+        check_response(res)
+        if wait_for_commit:
+            rev, term = extract_rev_term(res)
+            self.wait_for_commit(term, rev)
+        if check_receipt:
+            res_pb = ParseDict(res.json(), etcd_pb2.DeleteRangeResponse())
+            self.check_receipt("delete_range", req, res_pb)
+        return res
+
+    def get_receipt(self, rev: int, term: int):
+        """
+        Get a receipt for a revision and term.
+        """
+        logger.info("GetReceipt: {} {}", rev, term)
+        req = lskvserver_pb2.GetReceiptRequest()
+        req.revision = rev
+        req.raft_term = term
+        j = MessageToDict(req)
+        res = self.client.post(
+            "/v3/receipt/get_receipt",
+            json=j,
+        )
+        if res.status_code == http.HTTPStatus.ACCEPTED:
+            logger.info("GetReceipt: ACCEPTED")
+            # accepted, retry
+            res = self.client.post(
+                "/v3/receipt/get_receipt",
+                json=j,
+            )
+        return res
 
     def raw(self) -> httpx.Client:
         """
         Get the raw client.
         """
         return self.client
+
+    def check_receipt(self, req_type: str, request, response):
+        """
+        Check a receipt for a request and response.
+        """
+        rev, term = extract_rev_term_pb(response)
+        res = self.get_receipt(rev, term)
+        check_response(res)
+        body = res.json()
+        assert "receipt" in body
+        receipt = body["receipt"]
+        assert "txReceipt" in receipt
+        tx_receipt = receipt["txReceipt"]
+        assert "leafComponents" in tx_receipt
+        leaf_components = tx_receipt["leafComponents"]
+        assert "claimsDigest" in leaf_components
+        claims_digest = leaf_components["claimsDigest"]
+
+        response.ClearField("header")
```

Comment:
This is needed since the receipt contains responses before they have their header filled with data.